### PR TITLE
Adds option to control which connections to log (backport to v5/master)

### LINKED ISSUE #632: ApiQueryLogListener enables DebugKit query logs
Currently ApiQueryLogListener enables queryLogging in **ALL** connections, this includes DebugKit.

If you also have queries log enabled to file, it will fill with DebugKit unnecesary huge queries (panels, history, variables, etc), making it difficult to follow requests/cli queries.

Maybe it would be better to honor the *log* option in connection configuration, or to have a whitelist/blacklist on witch connection to log. Or simply ignore debug_kit logging

### PR DESCRIPTION
**Pull request by Phally**
Fixes #632

Backport to v5/master.

### CODE REVIEW

File: tests/TestCase/Listener/ApiQueryLogListenerTest.php [hunk 1]
```diff
@@ -198,6 +198,71 @@ public function testSetupLogging()
...
+     */
+    public function testSetupLoggingConfiguredSources()
+    {
+        $methodName = 'enableQueryLogging';
+        if (version_compare(Configure::version(), '3.7.0RC', '<')) {
+            $methodName = 'logQueries';
+        }
+
+        $DefaultSource = $this
+            ->getMockBuilder('\Cake\Database\Connection')
```
Comment (ADmad) [inline tests/TestCase/Listener/ApiQueryLogListenerTest.php:214]: Please `Connection::class` instead of string, same for other classes.

Comment (Phally) [reply tests/TestCase/Listener/ApiQueryLogListenerTest.php:214]: Haha I just copied it from an existing test case above. Will update both.

Comment (ADmad) [reply tests/TestCase/Listener/ApiQueryLogListenerTest.php:214]: I you want you can update existing ones too :slightly_smiling_face:

Comment (Phally) [reply tests/TestCase/Listener/ApiQueryLogListenerTest.php:214]: Yeah that is the plan. Going to look through the plugin and fix them.

**Comment by Phally** [comment]
Okay, I find/replace-ed the string class names per your request 😄

**Comment by ADmad** [comment]
Please add info about this features to the docs at `docs/listeners/api-query-log.rst`

**Comment by Phally** [comment]
> Please add info about this feature to the docs at `docs/listeners/api-query-log.rst`

Oops forgot! I've added it in both.

**Comment by Phally** [comment]
@ADmad Because Stickler does a correction, codesniffer now fails.

How am I supposed to fix this? Stickler will keep doing the correction.

They have a different opinion what alphabetically exactly is.. 😒

File: .stickler.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
 linters:
   phpcs:
-    standard: CakePHP
+    standard: CakePHP3
```
Comment (ADmad) [inline .stickler.yml:4]: It's called `CakePHP3` now? Don't remember such a change being done for core.

Comment (Phally) [reply .stickler.yml:4]: Well that's exactly where I stole it from:

https://github.com/cakephp/cakephp/commit/8a10d4e81ed190d8129e0c1be07dfb93b2732a78

**Pull request merged**